components/email.php

```diff
@@ -0,0 +1,213 @@
+
+		$defaults = array(
+			'post_status' => 'publish',
+			'post_type'   => bp_get_email_post_type(),
+		);
+
+		$email = array(
+			'post_title'   => $assoc_args['subject'],
+			'post_content' => $assoc_args['content'],
+			'post_excerpt' => ! empty( $assoc_args['plaintext_content'] ) ? $assoc_args['plaintext_content'] :'',
+		);
+
+		// Email post content.
+		$post_id = wp_insert_post( bp_parse_args( $email, $defaults, 'install_email_' . $id ) );
+
+		// Save the situation.
+		if ( ! is_wp_error( $post_id ) ) {
+			$tt_ids = wp_set_object_terms( $post_id, $id, bp_get_email_tax_type() );
+
+			// Situation description.
+			if ( ! is_wp_error( $tt_ids ) && ! empty( $assoc_args['type_description'] ) ) {
+				$term = get_term_by( 'term_taxonomy_id', (int) $tt_ids[0], bp_get_email_tax_type() );
+				wp_update_term( (int) $term->term_id, bp_get_email_tax_type(), array(
+					'description' => $assoc_args['type_description'],
+				) );
+			}
+
+			WP_CLI::success( "Email post created for type '{$assoc_args['type']}'." );
+		} else {
+			WP_CLI::error( "There was a problem creating the email post for type '{$assoc_args['type']}'." );
+		}
+
+		if ( true === $switched ) {
+			restore_current_blog();
+		}
+	}
+
+	/**
+	 * Get details for a post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <type>
+	 * : The email type to fetch the post details for.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole post, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields. Defaults to all fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     # Output the post ID for the 'activity-at-message' email type
+	 *     $ wp bp email get activity-at-message --fields=ID
+	 *
+	 * @subcommand get-post
+	 */
+	public function get_post( $args, $assoc_args ) {
+		$email = bp_get_email( $args[0] );
+
+		if ( is_wp_error( $email ) ) {
+			WP_CLI::error( "Email post for type '{$args[0]}' does not exist." );
+			return;
+		}
+
+		$post_arr = get_object_vars( $email->get_post_object() );
+		unset( $post_arr['filter'] );
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( $post_arr );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+		$formatter->display_item( $email->get_post_object() );
+	}
+
+	/**
+	 * Reinstall BuddyPress default emails.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--yes]
+	 * : Answer yes to the confirmation message.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp email reinstall --yes
+	 *     Success: Emails have been successfully reinstalled.
+	 */
+	public function reinstall( $args, $assoc_args ) {
+		WP_CLI::confirm( 'Are you sure you want to reinstall BuddyPress emails?', $assoc_args );
+
+		require_once buddypress()->plugin_dir . 'bp-core/admin/bp-core-admin-tools.php';
```

Comment:
I'd rather load it like in the tools component.

```
WP_CLI::add_command( 'bp tool', 'BPCLI_Tool', array(
	'before_invoke' => function() {
		require_once( buddypress()->plugin_dir . 'bp-core/admin/bp-core-admin-tools.php' );
	},
) );
```

Reply:
I thought about that, but the `wp bp email reinstall` sub-command is only used once.  The other sub-commands do not require the admin code, so I'd prefer that we do not use `before_invoke` here unless we can do it on a sub-command basis.

Reply:
That's true! :)